### DISCUSSION TITLE
How do you stop a tinyhttp server?

### DISCUSSION THREAD

**Discussion by adidahiya** [talentlessguy/tinyhttp#424] [Q&A]
How do I stop listening on a port after I've started my server? In Express there seems to be a `server.close()` function (see [graceful shutdown docs](https://expressjs.com/en/advanced/healthcheck-graceful-shutdown.html#graceful-shutdown)) but I don't see anything related to this in tinyhttp.

[`node:http`](https://nodejs.org/api/http.html#http_server_close_callback) also has a server close method.

**Comment by v1rtl** [comment]
Easiest way is this:

```js
import { App } from '@tinyhttp/app'
import { createServer } from 'http'

const app = new App()

const server = createServer((req, res) => {
  app.attach(req, res)
}).listen(3000)

if (something) server.close()
```

**Reply by adidahiya** [reply]
Hmm, that doesn't seem to type check properly:

![image](https://github.com/tinyhttp/tinyhttp/assets/723999/a1bd92cf-3edf-4812-9ace-b01ee69b039d)

I'm using @tinyhttp/app v2.2.1 and @types/node v20.10.5

**Reply by aarontravass** [reply]
Maybe try with `req as Request`?

**Reply by v1rtl** [reply]
the type error happens because `Request` of tinyhttp and node one are different types. Easiest solution is `req: Request`.
